backend/src/users/user.service.ts

```diff
@@ -22,14 +22,33 @@ export class UsersService {
         return this.userModel.find().exec();
     }
 
-    async findOne(@Param(':id') id: number | string) {
+    async findById(@Param(':id') id: number | string) {
         const users = await this.userModel.find().exec()
-        const user = users.filter(user => user._id.toString() === id)
-        return user
+        return users.filter(user => user._id.toString() === id)
     }
 
-    async updateById(@Param(':id') id, updateUser) {
-        const updateUserDto = await this.findOne(id)
+    async login(@Param() params: LoginUserDto) {
+        if (!params.password || !params.email) {
+            return null
+        }
+
+        try {
+            const user = await this.userModel.findOne({email: params.email})
+            const isPasswordCorrect = await bcrypt.compare(params.password, user.password)
```

Comment:
You can exit early here, if `isPasswordCorrect` is `false` here throw error.

If not continue with token creation.